webapp/src/components/markdownEditorInput/markdownEditorInput.tsx

```diff
@@ -0,0 +1,149 @@
+import Editor from '@draft-js-plugins/editor'
+import createMentionPlugin, {
+    defaultSuggestionsFilter,
+    MentionData,
+} from '@draft-js-plugins/mention'
+import '@draft-js-plugins/mention/lib/plugin.css'
+import './markdownEditorInput.scss'
+
+import createEmojiPlugin from '@draft-js-plugins/emoji'
+import '@draft-js-plugins/emoji/lib/plugin.css'
+
+import {getWorkspaceUsersList} from '../../store/users'
+import {useAppSelector} from '../../store/hooks'
+import {IUser} from '../../user'
+
+const imageURLForUser = (window as any).Components?.imageURLForUser
+
+type Props = {
+    onChange?: (text: string) => void
+    onFocus?: () => void
+    onBlur?: (text: string) => void
+    initialText?: string
+    id?: string
+}
+
+const MarkdownEditorInput = (props: Props): ReactElement => {
+    const {onChange, onFocus, onBlur, initialText, id} = props
+    const workspaceUsers = useAppSelector<IUser[]>(getWorkspaceUsersList)
+    const mentions: MentionData[] = useMemo(() => workspaceUsers.map((user) => ({name: user.username, avatar: `${imageURLForUser ? imageURLForUser(user.id) : ''}`})), [workspaceUsers])
+    const ref = useRef<Editor>(null)
+    const [editorState, setEditorState] = useState(() => {
+        const state = EditorState.moveFocusToEnd(EditorState.createWithContent(ContentState.createFromText(initialText || '')))
+        return EditorState.moveSelectionToEnd(state)
+    })
+    const [isMentionPopoverOpen, setIsMentionPopoverOpen] = useState(false)
+    const [isEmojiPopoverOpen, setIsEmojiPopoverOpen] = useState(false)
+    const [suggestions, setSuggestions] = useState(mentions)
+
+    const {MentionSuggestions, plugins, EmojiSuggestions} = useMemo(() => {
+        const mentionPlugin = createMentionPlugin({mentionPrefix: '@'})
+        const emojiPlugin = createEmojiPlugin()
+
+        // eslint-disable-next-line no-shadow
+        const {EmojiSuggestions} = emojiPlugin
+        // eslint-disable-next-line no-shadow
+        const {MentionSuggestions} = mentionPlugin
+        // eslint-disable-next-line no-shadow
+        const plugins = [
+            mentionPlugin,
+            emojiPlugin,
+        ]
+        return {plugins, MentionSuggestions, EmojiSuggestions}
+    }, [])
+
+    useEffect(() => {
+        if (ref) {
+            setTimeout(() => ref.current?.focus(), 200)
+        }
+    })
+
+    const customKeyBindingFn = useCallback((e: React.KeyboardEvent) => {
+        if (isMentionPopoverOpen || isEmojiPopoverOpen) {
+            return undefined
+        }
+
+        if (e.key === 'Escape') {
+            return 'editor-blur'
+        }
+
+        return getDefaultKeyBinding(e as any)
+    }, [isEmojiPopoverOpen, isMentionPopoverOpen])
+
+    const handleKeyCommand = (command: string): DraftHandleValue => {
+        if (command === 'editor-blur') {
+            ref.current?.blur()
+            return 'handled'
+        }
+
+        return 'not-handled'
+    }
+
+    const onEditorStateBlur = () => {
```

Comment:
I meant all the callbacks, because there are multiple of them, and you are passing a lot of them to the editor.

Reply:
Sorry I missed that. I've used it on all callbacks. 

Please double check to make sure that I've not missed any dependency on the callbacks. Thank you